### ISSUE TITLE
Support for VS 2017?

### ISSUE THREAD

**Issue by SimplerSoftware** [angular/vscode-ng-language-service#92]
Is there any chance that VS2017 will become a supported IDE?

**Comment by meriturva** [comment]
same request here! just to know if we could still on our IDE or move to vscode. Thanks!

**Comment by kukjevov** [comment]
i think if you set up typescript plugin to @angular/language-services it will work also in Visual Studio 2017, but not in standalone html, for that we will probably need plugin, but I am not completely sure, because @chuckjaz was doing also some changes in typescript to locate external html files for typescript.

**Comment by chuckjaz** [comment]
There are no plans for this extension to support VS2017.

However, from what I understand, there are changes coming to VS that would allow the language service to work as a plugin to TypeScript.

**Comment by siegerx3** [comment]
@chuckjaz 
Do you have any reference for that information?

**Comment by siegerx3** [comment]
@kukjevov 

> i think if you set up typescript plugin to @angular/language-services

What do you mean by that? Can you add plugins to the normal typescript plugin in vs2017?

**Comment by josephliccini** [comment]
It seems to be working if one includes it in `plugins` in the `tsconfig.json` in VS2017 15.4.

I don't see the language service working properly in our Angular HTML Templates though.

Here's the issue I filed: https://github.com/angular/angular/issues/19629

**Comment by SimplerSoftware** [comment]
Looks like Visual Studio is going to support the Angular language service natively.  See the following [announcement](https://blogs.msdn.microsoft.com/visualstudio/2017/10/24/visual-studio-2017-version-15-5-preview-2/?utm_source=vs_developer_news&utm_medium=referral) regarding VS 2017 15.5.  Here's a link to the [GitHub sample project](https://github.com/Microsoft/angular2-in-visual-studio)

**Comment by Narvalex** [comment]
Please do not forsake full Visual Studio... I do not like VS Code because it can not simply support properly multi-monitor setup

**Comment by manoharreddyporeddy** [comment]
Few things I found:

#1
angular website proposes to upgrade within an year.

Now that we have angular6 ready & 7 coming in a month

that makes this link (https://github.com/Microsoft/angular2-in-visual-studio) which is for angular 2, out of date.

#2
There is an article for VS 2015 on angular website here:
https://angular.io/guide/visual-studio-2015

Similar procedure for VS 2015 may work for VS 2017 with some changes

#3
Angular service (code completion and for debugging) is only available
for Visual Studio Code and WebStorm only
https://angular.io/guide/language-service#angular-language-service-in-your-editor

In other words, VS Code is not good for non-JS related works (like C++17) but it is good for JS & related works

After much thought & struggle, I learnt shortcuts of VS Code (https://code.visualstudio.com/assets/docs/getstarted/tips-and-tricks/KeyboardReferenceSheet.png), and started using it for Angular for now.

**Comment by Narvalex** [comment]
Visual studio code is not good for 3 or 2 monitors set up. Multi windows is not possible yet AFAIK

**Comment by manoharreddyporeddy** [comment]
True.

I switched to React.js for UI.

Angular2/4/5/6 is too difficult to get immediately started for my team.

Same is asserted here:
https://stackshare.io/react
https://stackshare.io/angular-2

In the above, React looks better, except for the votes.

In Angular, it took 3 hours to write a button & call an API,
and still has a lot of errors in Browser console.

While in React.js, our basic app is ready in just 1 day.

**Comment by duki994** [comment]
https://docs.microsoft.com/en-gb/visualstudio/extensibility/adding-an-lsp-extension?view=vs-2017#getting-started

This is promising. Newest VS2017 15.8+ versions include native support for Language Server Protocol

**Comment by RyoukoKonpaku** [comment]
Tried it using ```VS Preview 15.9``` and it works, ```inline templates``` have intellisense and gives errors as expected on how VS Code does it (no syntax highlighting though, probably need a grammar for that). ```Html templates``` have no intellisense which I think would need an extension similar to VS Code, but overall promising.
